File: add/codesnippet/VisualBasic/p-system.uritemplate.pat_1.vb
        Dim template As UriTemplate = New UriTemplate("weather/{state}/{city}?forecast={day}")

        Dim prefix As Uri = New Uri("http://localhost")

        Console.WriteLine("PathSegmentVariableNames:")
        For Each name As String In template.PathSegmentVariableNames
            Console.WriteLine("     {0}", name)
        Next

File: add/codesnippet/CSharp/p-system.uritemplate.pat_1.cs
            UriTemplate template = new UriTemplate("weather/{state}/{city}?forecast={day}");

            Uri prefix = new Uri("http://localhost");

            Console.WriteLine("PathSegmentVariableNames:");
            foreach (string name in template.PathSegmentVariableNames)
            {
                Console.WriteLine("     {0}", name);
            }